akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/MultipartUnmarshallers.scala

```diff
@@ -67,7 +70,7 @@ trait MultipartUnmarshallers {
               FastFuture.failed(new RuntimeException("Content-Type with a multipart media type must have a 'boundary' parameter"))
             case Some(boundary) ⇒
               import BodyPartParser._
-              val parser = new BodyPartParser(defaultContentType, boundary, log)
+              val parser = new BodyPartParser(defaultContentType, boundary, log, ParserSettings(ActorMaterializer.downcast(mat).system)) // TODO we could cache it
```

Comment:
Built like this, it isn't possible to override the settings for a particular use site. My (too brief) suggestion was to add an `parserSettingOverride: Option[ParserSetting]` parameter to all the methods that need it in MultipartUnmarshallers so that a user is able to create custom mutipart marshallers that use the custom setting.